### PR TITLE
Add some guidelines around mobile

### PR DESCRIPTION
**Pull request by mmislam101**

### CODE REVIEW

File: docs/Platforms.md [hunk 1]
```diff
@@ -76,15 +76,44 @@ A marketing site by its nature will have fewer moving parts than a full-fledged
...
-### Native Apps
+* [Fabric](https://get.fabric.io/) - Crash reporting
+* [Amplitude](https://amplitude.com/) - App metrics/events
+* [Realm](https://realm.io/) - Persistent local storage/database
 
+### iOS Native Apps
 
+The high adoption rate of new Apple technology, whether it's the latest devices or the latest OS, demands that we keep up with the latest tools and practices. We develop iOS applications using *XCode* with the latest version of *Swift*.
+
+Instead of a simple MVC Architecture, we've found that the MVVM fits better with iOS views and view controllers. MVVM allows us to create applications that are easy to read, and scales well with an Agile method of software development.
```
Comment (swilliams) [inline docs/Platforms.md]: Link to and/or explain what MVVM is.

File: docs/Platforms.md [hunk 2]
```diff
@@ -76,15 +76,44 @@ A marketing site by its nature will have fewer moving parts than a full-fledged
 
 We build mobile apps using native technologies, meaning that they interface directly with the libraries provided by the operating system, that being iOS and Android.
 
-We do not recommend mobile apps built in a web browser. The level of fit and finish is not the same as you get from native. We do however use cross-platform frameworks where appropriate.
+We do not recommend mobile apps built in a web browser. The level of fit and finish is not the same as you get from native. We do however use cross-platform frameworks where appropriate. For example, the following are some cross-platform libraries for the basic necessities of any mobile application:
 
-### Native Apps
+* [Fabric](https://get.fabric.io/) - Crash reporting
+* [Amplitude](https://amplitude.com/) - App metrics/events
+* [Realm](https://realm.io/) - Persistent local storage/database
```
Comment (swilliams) [inline docs/Platforms.md]: I think it'd be great to add a paragraph on the pros/cons of those. Fabric and Amplitude especially are big dependencies (file-size and perhaps even performance). Is there a scenario when it doesn't make sense to include those?

File: docs/Platforms.md [hunk 3]
```diff
@@ -76,15 +76,44 @@ A marketing site by its nature will have fewer moving parts than a full-fledged
...
 
+### iOS Native Apps
 
+The high adoption rate of new Apple technology, whether it's the latest devices or the latest OS, demands that we keep up with the latest tools and practices. We develop iOS applications using *XCode* with the latest version of *Swift*.
+
+Instead of a simple MVC Architecture, we've found that the MVVM fits better with iOS views and view controllers. MVVM allows us to create applications that are easy to read, and scales well with an Agile method of software development.
+
+For package and version control, we use Cocoapods, which is the most common method for developers to publish their iOS code (as of this writing). Below are some pods that we use as the foundation of our mobile applications:
```
Comment (swilliams) [inline docs/Platforms.md]: Link to CocoaPods and capitalize it properly ;)

File: docs/Platforms.md [hunk 4]
```diff
@@ -76,15 +76,44 @@ A marketing site by its nature will have fewer moving parts than a full-fledged
...
+
+* [RealmSwift](https://realm.io/docs/swift/latest/) - Realm offers a SQL-type store, similar to Core Data. It solves concurrency, like Core Data, with `Realm`s instead of `Context`s, but with a much easier interface and a simpler solution for creating and managing models. It also offers smooth decoding of JSON into a Realm objects, as long as the keys map directly.
+* [Moya](https://github.com/Moya/Moya) - Moya abstracts the network layer into a set of simple Swift config structs, where you define your routers, their HTTP methods, and even test/mock responses that you can unit test. It uses Alamofire as its core network layer.
+* [Locksmith](https://github.com/matthewpalmer/Locksmith) - Super simple library to abstract all your authentication keychain stuff.
+* [CocoaLumberjack/Swift](https://github.com/CocoaLumberjack/CocoaLumberjack) - This is a life savior for logging debug info into an emailable text file. Great for those production bugs that are hidden deep in background executions, or the ones that the crash reports provides *zero* help with.
```
Comment (swilliams) [inline docs/Platforms.md]: "life saver"

File: docs/Platforms.md [hunk 5]
```diff
@@ -76,15 +76,44 @@ A marketing site by its nature will have fewer moving parts than a full-fledged
...
 ### Cross Platform Apps
 
-### Native or Cross Platform?
+Facebook's [React-native](https://facebook.github.io/react-native/) currently offers the best cross-platform solution to develop iOS and Android applications, using the same-ish code base. Same-*ish* because you can easily create an entirely different experience for your Android application than your iOS application. Each platform comes with their own `index.js` file, and they have the *option* of sharing components and behaviors.
+
+The power of React-native is that you write interfaces that are translated to use the best native solutions under the hood. For example, iOS applications can use [NavigatorIOS](https://facebook.github.io/react-native/docs/navigatorios.html) which uses `UINavigationController` to give the same feel of a native iOS application.
+
+When you compile your react-native code base, a project for both iOS and Android is provided. The common code is compiled into libraries that each project consumes. You can inter-mix natively written iOS or Android components, along with components created through react-native.
+
+You'll also need either XCode or Android Studio to build and release the final product.
```
Comment (swilliams) [inline docs/Platforms.md]: Do we have recommended React Native libraries? I know we're using [NativeBase](https://nativebase.io) on one project.

File: docs/Platforms.md [hunk 6]
```diff
@@ -76,15 +76,44 @@ A marketing site by its nature will have fewer moving parts than a full-fledged
...
+* [Moya](https://github.com/Moya/Moya) - Moya abstracts the network layer into a set of simple Swift config structs, where you define your routers, their HTTP methods, and even test/mock responses that you can unit test. It uses Alamofire as its core network layer.
+* [Locksmith](https://github.com/matthewpalmer/Locksmith) - Super simple library to abstract all your authentication keychain stuff.
+* [CocoaLumberjack/Swift](https://github.com/CocoaLumberjack/CocoaLumberjack) - This is a life savior for logging debug info into an emailable text file. Great for those production bugs that are hidden deep in background executions, or the ones that the crash reports provides *zero* help with.
 
 ### Cross Platform Apps
 
-### Native or Cross Platform?
+Facebook's [React-native](https://facebook.github.io/react-native/) currently offers the best cross-platform solution to develop iOS and Android applications, using the same-ish code base. Same-*ish* because you can easily create an entirely different experience for your Android application than your iOS application. Each platform comes with their own `index.js` file, and they have the *option* of sharing components and behaviors.
```
Comment (swilliams) [inline docs/Platforms.md]: "React Native" (no hyphen)

File: docs/Platforms.md [hunk 7]
```diff
@@ -76,15 +76,44 @@ A marketing site by its nature will have fewer moving parts than a full-fledged
...
+
+You'll also need either XCode or Android Studio to build and release the final product.
+
+### Native or React-native?
+
+Ideally, if you have two developers who are each strong in either iOS or Android, they can collaborate on the same codebase, sharing the same components, while maintaining the UX that is best for each platform. Usually the problem of any cross-platform solution is that Android users have to deal with iOS type elements, and vice-versa, providing a *worst of both worlds* outcome for users. React-native allows the developers to code different UX (such as navigation) for their Android or iOS users, while still sharing as much of the common elements as they can.
+
+But, there is an overhead cost associated with using react-native. Where IDEs such as XCode and Android Studio has had years to optimize their toolset to solve all the common issues with building and releasing products, react-native forces developers to have to deal with npm packages, write code in simpler text editors such as Sublime, and all these need to be customized and optimized manually by the developers.
```
Comment (swilliams) [inline docs/Platforms.md]: "has had" -> "have had"

File: docs/Platforms.md (hunk 2 above)
Comment (mmislam101) [inline docs/Platforms.md]: 👍

File: docs/Platforms.md (hunk 1 above)
Comment (mmislam101) [inline docs/Platforms.md]: 👍

File: docs/Platforms.md (hunk 3 above)
Comment (mmislam101) [inline docs/Platforms.md]: 👍

File: docs/Platforms.md (hunk 6 above)
Comment (mmislam101) [inline docs/Platforms.md]: 👍

File: docs/Platforms.md (hunk 4 above)
Comment (mmislam101) [inline docs/Platforms.md]: 👍

File: docs/Platforms.md (hunk 5 above)
Comment (mmislam101) [inline docs/Platforms.md]: I'll wait until I (or someone else) has more experience and then add a proper React Native and Android section with recommended libraries and etc.

File: docs/Platforms.md (hunk 7 above)
Comment (mmislam101) [inline docs/Platforms.md]: 👍

**Comment by mmislam101** [comment]
@swilliams made updates

File: docs/Platforms.md [hunk 8]
```diff
@@ -79,25 +79,28 @@ We build mobile apps using native technologies, meaning that they interface dire
...
+Facebook's [React Native](https://facebook.github.io/react-native/) currently offers the best cross-platform solution to develop iOS and Android applications, using the same-ish code base. Same-*ish* because you can easily create an entirely different experience for your Android application than your iOS application. Each platform comes with their own `index.js` file, and they have the *option* of sharing components and behaviors.
 
 The power of React-native is that you write interfaces that are translated to use the best native solutions under the hood. For example, iOS applications can use [NavigatorIOS](https://facebook.github.io/react-native/docs/navigatorios.html) which uses `UINavigationController` to give the same feel of a native iOS application.
```
Comment (swilliams) [inline docs/Platforms.md]: Just do a find/replace on all `React-native` strings please 😺

**Comment by mmislam101** [comment]
@swilliams fixed all react-native

**Pull request merged**